dagutils/traversal.go

```diff
@@ -0,0 +1,384 @@
+package dagutils
+
+import (
+	"context"
+	"errors"
+
+	ipld "gx/ipfs/QmZtNq8dArGfnpCZfx2pUNY7UcjGhVp5qqwQ4hH6mpTMRQ/go-ipld-format"
+)
+
+// Traversal provides methods to move through a DAG of IPLD nodes
```

Comment:
Yes, rereading https://github.com/ipfs/go-ipfs/issues/5304#issue-345503022 I see that why's mentioning to actually move these type of packages there instead of `go-merkledag`.